### PR TITLE
CSHARP-2295: Deprecate group command helpers

### PR DESCRIPTION
**Pull request by ksadoff**
https://jira.mongodb.org/browse/CSHARP-2295
Updated June 11: https://evergreen.mongodb.com/version/5cfffdfac9ec441465cc1d61

### CODE REVIEW

**Comment by ksadoff** [comment]
Ready for interim review

**Review by vincentkam — CHANGES_REQUESTED** [review]
PR #2! 😸 A good portion of the comments I made in https://github.com/ksadoff/mongo-csharp-driver/pull/1 should also apply here

**Comment by ksadoff** [comment]
https://evergreen.mongodb.com/version/5cf9302f2fbabe336e1e86eb

File: tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs [hunk 1]
```diff
@@ -2150,6 +2156,8 @@ public void TestGroupWithKeyFunction()
         [SkippableFact]
         public void TestGroupWithMaxTime()
         {
+#pragma warning disable 618
```
Comment (vincentkam) [inline tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs]: This works! We tend use wrap only the deprecated method call with the `#pragma` statements though, like so: 
```
#pragma warning disable 618
                        Assert.Throws<MongoExecutionTimeoutException>(() => _collection.Group(args));
#pragma warning restore
```

This wasn't in our dev/style guide, so I've updated the guide to indicate this :)

File: tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs [hunk 2]
```diff
@@ -2073,15 +2073,17 @@ public void TestGroupWithFinalizeFunction()
             _collection.Insert(new BsonDocument("x", 3));
             _collection.Insert(new BsonDocument("x", 3));
             _collection.Insert(new BsonDocument("x", 3));
-
+
```
Comment (vincentkam) [inline tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs]: Super minor nit: there's some extraneous whitespace before and after the `#pragma` statements in this file that could be removed :)

**Review by vincentkam — CHANGES_REQUESTED** [review]
Getting close! 
@rstam Thoughts on updating `Feature.GroupCommand`'s `supportRemovedVersion`? I defined it as 4.0.0 rc1 in [CSHARP-2285](https://jira.mongodb.org/browse/CSHARP-2285), which is likely the first time we noticed when the server stopped responding to the command. The [driver ticket](https://jira.mongodb.org/browse/DRIVERS-370) states it was finally removed in 4.1.1. I see no harm in continuing to test it on the versions of the server that do support the group command, and the deprecation notices that Katie added correctly state that the `group` command was deprecated in server version 3.4.

**Review by rstam — APPROVED** [review]

**Comment by vincentkam** [comment]
@ksadoff : After a quick offline discussion with @rstam, we think it shouldn't hurt to update `Feature.GroupCommand`'s `supportRemovedVersion` to 4.1.1 as suggested by the Drivers ticket :)

File: src/MongoDB.Driver.Core/Core/Misc/Feature.cs [hunk 3]
```diff
@@ -53,7 +53,7 @@ public class Feature
         private static readonly Feature __findAndModifyWriteConcern = new Feature("FindAndModifyWriteConcern", new SemanticVersion(3, 2, 0));
         private static readonly Feature __findCommand = new Feature("FindCommand", new SemanticVersion(3, 2, 0));
         private static readonly Feature __geoNearCommand = new Feature("GeoNearCommand", new SemanticVersion(1, 0, 0), new SemanticVersion(4, 1, 0, ""));
-        private static readonly Feature __groupCommand = new Feature("GroupCommand", new SemanticVersion(1, 0, 0), new SemanticVersion(4, 0, 0, "rc1"));
+        private static readonly Feature __groupCommand = new Feature("GroupCommand", new SemanticVersion(1, 0, 0), new SemanticVersion(4, 1, 1, "rc1"));
```
Comment (vincentkam) [inline src/MongoDB.Driver.Core/Core/Misc/Feature.cs]: Sorry, I was unclear here!  We don't know the pre-release version where the command was removed so for now let's assume it was removed in 4.1.1 with a pre-release string of `""`, which effectively says "any pre release version of 4.1.1" (see [here ](https://github.com/mongodb/mongo-csharp-driver/blob/21aff044981244cf1158c6d8e01e6bf542a3ed17/src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs#L107)for all the gory details).

**Review by vincentkam — CHANGES_REQUESTED** [review]
Almost there! 😄

**Review by vincentkam — APPROVED** [review]
LGTM 😄

**Comment by ksadoff** [comment]
Ready for final review

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Comment by ksadoff** [comment]
Rebased, merged, and pushed.

**Pull request closed without merging**